ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -82,12 +82,70 @@ type domain_info = {
   ; machine_pwd_last_change_time: float option
 }
 
-let hd msg = function
-  | [] ->
-      error "%s" msg ;
-      raise (Auth_service_error (E_GENERIC, msg))
-  | h :: _ ->
-      h
+let generic_error msg =
+  error "%s" msg ;
+  raise (Auth_service_error (E_GENERIC, msg))
+
+let fail fmt = Printf.ksprintf generic_error fmt
+
+(** Kerberos Domain Controller. The current implementation does not
+    work with non-standard ports *)
+module KDC : sig
+  type t
+
+  val server : t -> string
```

Comment:
is this some kind of convention? Why not call it something like `address` or `ip_addr`?

Reply:
This is nitpicking, but I think following Uri's convention of `host` would make sense

Reply:
because it's used as the `--server`parameter for queries

Reply:
The `server` name matches the `-server` option where it is used.